File: Easy/Range Sum of BST/range-sum-of-bst.py
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:

        s = 0
        stack = [ root ]

        while stack:
            n = stack.pop()

            if low <= n.val <= high:
                s += n.val

            if n.left is not None: stack.append(n.left)
            if n.right is not None: stack.append(n.right)
        
        return s

File: Easy/Range Sum of BST/range-sum-of-bst.cpp
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        
        uint sum = 0;
        stack<TreeNode*> stack;
        stack.push(root);

        while(!stack.empty())
        {
            auto node = stack.top();
            stack.pop();

            if(node->val >= low && node->val <= high)
                sum += node->val;

            if(node->left != nullptr) stack.push(node->left);
            if(node->right != nullptr) stack.push(node->right);
        }

        return sum;
    }
};
